### PR TITLE
fix(table-of-contents): TOC scroll-behavior style typo

### LINKED ISSUE #10493: [TOC bug]: TOC un-smooth scroll behavior
### Description

I noticed a problem with the Table of contents component, it has transition on the [storybook example](https://www.ibm.com/standards/carbon/web-components/react/?path=/story/components-table-of-contents--default), but when I tried it locally and [on sandbox](https://codesandbox.io/s/little-worker-t84isr?resolutionWidth=1540&resolutionHeight=675&file=/src/index.js:264-282), it has a different behavior.

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/41386480/123c1445-f8ed-4e1f-b4d8-2cef35fed244

A proposed solution was to add `html {
  scroll-behavior: smooth;
}` but it will affect all the scroll behaviors in the site.

This can be seen while tabbing through the cards as in the following video, the default behavior was to tab without smoothing.
I remember when we were using the TOC from `@carbon/ibmdotcom-react` it was smoothing only on the TOC and the rest of the website had its default behavior. But when we migrated to web-components we had this problem.
Another example why a applying global scroll behavior is not preferred, is if a page scrolls for any reason, like going to a specific anchor location, or a refresh, etc. it would have the global scroll behavior, which I think is not true.

Here's [another sandbox](https://codesandbox.io/p/sandbox/beautiful-resonance-qnylq3?file=%2Fsrc%2Findex.js%3A1%2C1) for the smooth behavior with card group component, the default behavior for tabbing was overridden with the global scroll behavior.

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/41386480/5a25e4f6-ae5b-489f-95dc-b55ee88c51a2

### Component(s) impacted

https://www.ibm.com/standards/carbon/web-components/react/?path=/story/components-table-of-contents--default

### Browser

Chrome

### Carbon for IBM.com version

v1.28.0

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

ibm.com/impact

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://codesandbox.io/p/sandbox/little-worker-t84isr?file=%2Fsrc%2Findex.css%3A12%2C13

### Steps to reproduce the issue (if applicable)

_No response_

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by raluca-danaila**
### Related Ticket(s)

Closes #10493 

https://jsw.ibm.com/browse/ADCMS-3701

### Description

I have fixed the typo in the table-of-contents from  `scroll-behaviour: smooth;` to  `scroll-behavior: smooth;`

### Changelog

**Changed**

- changed the mistyping: `scroll-behaviour: smooth;` to  `scroll-behavior: smooth;`  in the "table-of-contents" component.

### CODE REVIEW

**Review by annawen1 — APPROVED** [review]
LGTM! Thanks! 🎉

**Review by ariellalgilmore — APPROVED** [review]
Looks good! @raluca-danaila if you could just comment below the follow:

> I have read the DCO document and I hereby sign the DCO.

**Comment by raluca-danaila** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by ariellalgilmore** [comment]
recheck

**Comment by andy-blum** [comment]
If we want smooth scrolling to be the default we should probably also add that config in to the JS that makes the scrolling happen.

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/table-of-contents/table-of-contents.ts#L463

https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollTo#parameters

File: packages/web-components/src/components/table-of-contents/table-of-contents.ts [hunk 1]
```diff
@@ -462,6 +462,12 @@ class DDSTableOfContents extends HostListenerMixin(
 
       window.scrollTo(0, targetY);
```
Comment (andy-blum) [inline packages/web-components/src/components/table-of-contents/table-of-contents.ts:463]: We have to remove the old scroll behavior for the new one to appear.

**Review by andy-blum — CHANGES_REQUESTED** [review]

File: packages/styles/scss/components/tabs-extended/_tabs-extended.scss [hunk 2]
```diff
@@ -206,6 +206,16 @@
       }
     }
   }
+  #{$prefix}--accordion__content {
```
Comment (m4olivei) [inline packages/styles/scss/components/tabs-extended/_tabs-extended.scss:209]: This hunk looks like an unintentional change that came in with commit https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/10840/commits/323c87167fc803cd51aa117b54e4cc4e89355dd1 . @marcelojcs was this intentional? 

If it is meant to be here, the opening and closing braces are unbalenced, which causes failures in the CI jobs.

File: packages/web-components/src/components/table-of-contents/table-of-contents.ts [hunk 3]
```diff
@@ -460,7 +460,11 @@ class DDSTableOfContents extends HostListenerMixin(
         targetY = elem.offsetTop - mobilePadding!;
       }
 
-      window.scrollTo(0, targetY);
+      window.scrollTo({
+        top:targetY,
```
Comment (m4olivei) [inline packages/web-components/src/components/table-of-contents/table-of-contents.ts:464]: Nit pick.
```suggestion
        top: targetY,
```

File: packages/web-components/src/components/table-of-contents/table-of-contents.ts [hunk 4]
```diff
@@ -460,7 +460,11 @@ class DDSTableOfContents extends HostListenerMixin(
...
+        left: 0,
+        behavior:'smooth'
```
Comment (m4olivei) [inline packages/web-components/src/components/table-of-contents/table-of-contents.ts:466]: Another nit pick (sorry).
```suggestion
        behavior: 'smooth'
```

**Review by m4olivei — CHANGES_REQUESTED** [review]
There are some legit CI job failures that are introduced here. See comments.

**Comment by marcelojcs** [comment]
Closed in favor of
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/10982

**Pull request closed without merging**